tests/rptest/tests/cluster_linking_e2e_test.py

```diff
@@ -1109,6 +1109,41 @@ def test_replication_with_transactions(self):
         )
         self.verify()
 
+    @cluster(num_nodes=8)
+    def test_replication_with_truncated_topic(self):
```

Comment:
just to be clear, this reliably fails if you remove the previous commit, yeah?

Reply:
It reproduces pretty reliably (especially if we bump the msg_count further).. Check my other comment, it depends on broker responses for partitions not included in a particular fetch but in the session.